server/src/app.module.ts

```diff
@@ -2,43 +2,58 @@ import { Logger, Module } from "@nestjs/common";
 import { ScheduleModule } from "@nestjs/schedule";
 import { ServeStaticModule } from "@nestjs/serve-static";
 import { TypeOrmModule } from "@nestjs/typeorm";
-import * as path from "path";
 import { DataSource } from "typeorm";
 import { AppConfigModule } from "./config/app/configuration.module";
+import { AppConfigService } from "./config/app/configuration.service";
 import { DatabaseConfigModule } from "./config/database/configuration.module";
 import { DatabaseConfigService } from "./config/database/configuration.service";
 import { SchedulerConfigModule } from "./config/scheduler/configuration.module";
 import { Block } from "./database/entities/block.entity";
 import { Neighborhood } from "./database/entities/neighborhood.entity";
+import { TransactionDetails } from "./database/entities/transaction-details.entity";
 import { Transaction } from "./database/entities/transaction.entity";
 import { NeighborhoodModule } from "./neighborhoods/neighborhood.module";
 import { NetworkService } from "./services/network.service";
-import { SchedulerService } from "./services/scheduler.service";
-
-// The root path for all static files.
-const staticRootPath = path.join(__dirname, "../..", "client/build");
+import { SchedulerModule } from "./services/scheduler/scheduler.module";
 
 @Module({
   controllers: [],
-  providers: [Logger, NetworkService, SchedulerService],
+  providers: [Logger, NetworkService],
   imports: [
     AppConfigModule,
     NeighborhoodModule,
     SchedulerConfigModule,
     ScheduleModule.forRoot(),
-    ServeStaticModule.forRoot({
-      rootPath: staticRootPath,
+    SchedulerModule,
+    ServeStaticModule.forRootAsync({
```

Comment:
One of the reasons for switching to Vite is that I can build in watch mode (instead of just hot reload). I suspect we can make NestJS reload static assets when the build folder changes if it's also in watch mode.

Reply:
There's nothing to reload. The files are read from the disk everytime. This isn't meant to be optimized, which is also why I allowed use to remove the static serve module (in prod we'll want to have NGINX take the job of service static assets).

Reply:
Okay, then I should be able to get watch mode working pretty easily. Thanks!